trl/trainer/dpo_trainer.py

```diff
@@ -575,7 +578,10 @@ def _prepare_deepspeed(self, model: PreTrainedModelWrapper):
         # If ZeRO-3 is used, we shard both the active and reference model.
         # Otherwise, we assume the reference model fits in memory and is initialized on each device with ZeRO disabled (stage 0)
         if config_kwargs["zero_optimization"]["stage"] != 3:
-            config_kwargs["zero_optimization"]["stage"] = 0
+            if self.ref_model_use_stage_3:
+                config_kwargs["zero_optimization"]["stage"] = 3
+            else:
+                config_kwargs["zero_optimization"]["stage"] = 0
```

Comment:
Would you like to re-use the following function?

https://github.com/huggingface/trl/blob/ba6abee37f0f0463f6d891d63d0c2242039fc8ec/trl/trainer/utils.py#L1030